types/log/log.go

```diff
@@ -0,0 +1,58 @@
+package log
+
+import (
+	"context"
+	"time"
+
+	"github.com/globalsign/mgo/bson"
+	"github.com/tsuru/tsuru/types/auth"
+)
+
+type LogType string
+
+const (
+	LogTypeApp = LogType("app")
+	LogTypeJob = LogType("job")
+)
+
+type LogabbleObject interface {
+	GetName() string
+	GetPool() string
+}
+
+type LogWatcher interface {
+	Chan() <-chan LogEntry
+	Close()
+}
+
+type LogService interface {
+	Enqueue(entry *LogEntry) error
+	Add(name, tsuruType string, message, source, unit string) error
+	List(ctx context.Context, args ListLogArgs) ([]LogEntry, error)
+	Watch(ctx context.Context, args ListLogArgs) (LogWatcher, error)
+}
+
+type LogServiceProvision interface {
+	Provision(name, tsuruType string) error
+	CleanUp(name, tsuruType string) error
+}
+
+type ListLogArgs struct {
+	Name         string
+	Type         LogType
+	Source       string
+	Units        []string
+	Limit        int
+	InvertSource bool
+	Token        auth.Token
+}
+
+type LogEntry struct {
+	MongoID bson.ObjectId `bson:"_id,omitempty" json:"-"`
```

Comment:
Do we yet store the log entries in MongoDB? I really don't think so... Can check again? If we don't please remove this field.